flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -291,5 +325,34 @@ public void run(SourceContext<String> ctx) throws Exception {
 		@Override
 		public void cancel() {}
 	}
+
+	private static class TestBoundedSource implements SourceFunction<String> {
```

Comment:
I'm pretty sure there is already something like that defined somewhere. For example:
`org.apache.flink.test.streaming.runtime.StreamTaskSelectiveReadingITCase.TestStringSource`
`org.apache.flink.streaming.runtime.tasks.SourceStreamTaskTest.MockSource`
or even production class
`org.apache.flink.streaming.api.functions.source.FromElementsFunction`

Please re-use one of them, preferably `FromElementsFunction`.